### PR TITLE
Return first hit, simplify code

### LINKED ISSUE #267: Don't bother returning more than 1 hit
BlockchainIndex returns 10 hits, but only the first one is observed and returned. Simplify the code, return first hit only, and remove the weird for loops that returns on the first element.

### PR DESCRIPTION
**Pull request by sudo-update**
## PR description
"BlockchainIndex" returned 10 hits, but only the first one was observed and returned. I simplified the code to returned first hit only and removed the for loops.
## Fixed Issue(s)
fixes #267

### CODE REVIEW

File: eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt [hunk 1]
```diff
@@ -483,10 +483,8 @@ class BlockchainIndex(private val indexWriter: IndexWriter) : BlockchainIndexWri
       val topDocs = searcher!!.search(query, HITS)
 
       val docs = mutableListOf<Document>()
-      for (hit in topDocs.scoreDocs) {
-        val doc = searcher.doc(hit.doc, setOf("_id") + fields)
-        docs += doc
-      }
+      val doc = searcher.doc(topDocs.scoreDocs.elementAt(0).doc, setOf("_id") + fields)
+      docs += doc
```
Comment (atoulme) [inline eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt:487]: Maybe just return the doc instead of a list here.

**Comment by atoulme** [comment]
Can you also take care of line 554? 10 is inlined there, should be HITS. And there's a for loop right below.

File: eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt (hunk 1 above)
Comment (sudo-update) [reply eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt:487]: It will change the return type from List<Document> to a single Document, is that okay?

Comment (atoulme) [reply eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt:487]: Looks ok from here.

Comment (sudo-update) [reply eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt:487]: When I make the above change, I get this error:
Task :eth-repository:compileKotlin FAILED
e: /home/gradle/eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt: (485, 14): Type mismatch: inferred type is Document! but List<Document> was expected

FAILURE: Build failed with an exception.

**Comment by sudo-update** [comment]
> Can you also take care of line 554? 10 is inlined there, should be HITS. And there's a for loop right below.

I have fixed the issue on line 554 as well.

However I cannot change the return type to a doc from a list (as requested) due to a build exception.

**Review by atoulme — APPROVED** [review]
lgtm

**Comment by sudo-update** [comment]
> lgtm

is there anything else I should do, in order to get this pull request merged?

**Comment by atoulme** [comment]
You have to get a successful build. Not sure why it fails right now.

**Comment by sudo-update** [comment]
I am unsure why, but GitHub was saying I was behind main, so I merged from main, just to make sure I was working from the latest codebase.  I have changed EthHandlerTest.kt to not loop through 10 indices. On my end, I am building successfully.

**Comment by sudo-update** [comment]
I am unable to start the workflow check, as a first time contributor

**Comment by atoulme** [comment]
Running the ci now.

File: devp2p-eth/src/test/kotlin/org/apache/tuweni/devp2p/eth/EthHandlerTest.kt [hunk 2]
```diff
@@ -128,22 +128,22 @@ class EthHandlerTest {
...
+      val newBlock = createChildBlock(header)
+      repository.storeBlock(newBlock)
+      var txIndex = 0
+
+      repository.storeTransactionReceipt(
+        TransactionReceipt(
+          Bytes32.random(),
+          32L, LogsBloomFilter(), emptyList()
+        ),
+        txIndex, newBlock.body.transactions.hash, newBlock.header.hash
```
Comment (atoulme) [inline devp2p-eth/src/test/kotlin/org/apache/tuweni/devp2p/eth/EthHandlerTest.kt:141]: this is not going to compile, the list of transactions doesn't have a hash, each transaction has its own hash.
Is there a reason for modifying this test? Seems unrelated to your changes.

File: eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt [hunk 3]
```diff
@@ -553,16 +550,16 @@ class BlockchainIndex(private val indexWriter: IndexWriter) : BlockchainIndexWri
...
         TermQuery(Term("_type", "difficulty")),
-        10,
+        HITS,
         Sort(SortField(TOTAL_DIFFICULTY.fieldName, SortField.Type.STRING, true))
       )
-
-      for (hit in topDocs.scoreDocs) {
-        val doc = searcher.doc(hit.doc, setOf("_id"))
-        val bytes = doc.getBinaryValue("_id")
+      val doc = searcher.doc(topDocs.scoreDocs.elementAt(0).doc, setOf("_id"))
```
Comment (atoulme) [inline eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt:556]: What if scoreDocs is empty? Would be good to check the element exists, and return null early otherwise.

Comment (sudo-update) [reply eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt:556]: ### Looking into it now

**Comment by sudo-update** [comment]
I think I have fixed the issue now, I am getting build successful on my end.  I have requested re-review, hopefully I have succeeded!

I reverted my changes to EthHandlerTest, you were correct.  I did not need to change anything there.  I also made the changes you suggested, to return null when scoreDocs was empty.  That solved the build failure, I think that was the issue.

**Comment by sudo-update** [comment]
Awesome! It passed all checks! Does it look good to you?

**Review by atoulme — APPROVED** [review]
LGTM
